### ISSUE TITLE
Reconsider deprecation process

### ISSUE THREAD

**Issue by michitux** [splitbrain/dokuwiki#2347]
According to the [deprecation policy](https://www.dokuwiki.org/devel:releases#deprecation), all interfaces that shall be removed are first deprecated and only removed after they have been deprecated in two releases. However, in the Greebo release, at least the action handling refactoring removed a lot of functions like `act_permcheck` that were not deprecated. This breaks at least the [saveandedit](https://www.dokuwiki.org/plugin:saveandedit) plugin. According to the documentation, this should not happen as, as far as I'm aware, the plugin does not use any deprecated functions. I'm well aware that the plugin depends a lot on very specific internals of dokuwiki, but still after reading the deprecation policy I would expect that functions like `act_permcheck` or `act_save` don't suddenly disappear from one release to the next.

Unless I'm mistaken and these functions were never supposed to be part of the API, I think either the deprecation policy should be changed or these functions should be restored in the form of compatibility wrappers.

**Comment by splitbrain** [comment]
Yeah, this is very tricky. The main problem we have is that it is not clear what can be considered public API and what is internals that may change without warning.

Ideally plugin authors would have an eye on the DokuWiki development and notice when pull requests remove functionality they rely on, but I know that that isn't the case.

Especially when we do bigger refactoring (which some parts of DokuWiki really need) it is hard to maintain backwards compatibility. I'm not sure we can come up with reasonable implementations for the functions mentioned by you. PR welcome (if it's simple it might be considered for an upcoming hotfix release)
